src/aliceVision/lensCorrectionProfile/lcp.cpp

```diff
@@ -0,0 +1,986 @@
+        pOut.ChromaticBlueGreenParams.RadialDistortParam1 = interpolate<float>(weightLow, p1.ChromaticBlueGreenParams.RadialDistortParam1, p2.ChromaticBlueGreenParams.RadialDistortParam1);
+        pOut.ChromaticBlueGreenParams.RadialDistortParam2 = interpolate<float>(weightLow, p1.ChromaticBlueGreenParams.RadialDistortParam2, p2.ChromaticBlueGreenParams.RadialDistortParam2);
+        pOut.ChromaticBlueGreenParams.RadialDistortParam3 = interpolate<float>(weightLow, p1.ChromaticBlueGreenParams.RadialDistortParam3, p2.ChromaticBlueGreenParams.RadialDistortParam3);
+        pOut.ChromaticBlueGreenParams.isEmpty = false;
+        pOut.ChromaticRedGreenParams.FocalLengthX = interpolate<float>(weightLow, p1.ChromaticRedGreenParams.FocalLengthX, p2.ChromaticRedGreenParams.FocalLengthX);
+        pOut.ChromaticRedGreenParams.FocalLengthY = interpolate<float>(weightLow, p1.ChromaticRedGreenParams.FocalLengthY, p2.ChromaticRedGreenParams.FocalLengthY);
+        pOut.ChromaticRedGreenParams.ImageXCenter = interpolate<float>(weightLow, p1.ChromaticRedGreenParams.ImageXCenter, p2.ChromaticRedGreenParams.ImageXCenter);
+        pOut.ChromaticRedGreenParams.ImageYCenter = interpolate<float>(weightLow, p1.ChromaticRedGreenParams.ImageYCenter, p2.ChromaticRedGreenParams.ImageYCenter);
+        pOut.ChromaticRedGreenParams.RadialDistortParam1 = interpolate<float>(weightLow, p1.ChromaticRedGreenParams.RadialDistortParam1, p2.ChromaticRedGreenParams.RadialDistortParam1);
+        pOut.ChromaticRedGreenParams.RadialDistortParam2 = interpolate<float>(weightLow, p1.ChromaticRedGreenParams.RadialDistortParam2, p2.ChromaticRedGreenParams.RadialDistortParam2);
+        pOut.ChromaticRedGreenParams.RadialDistortParam3 = interpolate<float>(weightLow, p1.ChromaticRedGreenParams.RadialDistortParam3, p2.ChromaticRedGreenParams.RadialDistortParam3);
+        pOut.ChromaticRedGreenParams.isEmpty = false;
+    }
+    }
+}
+
+void LCPinfo::getDistortionParams(const float& focalLength, const float& focusDistance, LensParam& lparam)
+{
+    settingsInfo userSettings;
+    userSettings.ApertureValue = 0.f;
+    userSettings.FocalLength = focalLength;
+    userSettings.FocusDistance = focusDistance;
+
+    int iLow, iHigh;
+    float weightLow;
+    if (search(userSettings, LCPCorrectionMode::DISTORTION, iLow, iHigh, weightLow))
+    {
+        combine(iLow, iHigh, weightLow, LCPCorrectionMode::DISTORTION, lparam);
+    }
+}
+
+void LCPinfo::getVignettingParams(const float& focalLength, const float& aperture, LensParam& lparam)
+{
+    settingsInfo userSettings;
+    userSettings.ApertureValue = aperture;
+    userSettings.FocalLength = focalLength;
+    userSettings.FocusDistance = 0.f;
+
+    int iLow, iHigh;
+    float weightLow;
+    if (search(userSettings, LCPCorrectionMode::VIGNETTE, iLow, iHigh, weightLow))
+    {
+        combine(iLow, iHigh, weightLow, LCPCorrectionMode::VIGNETTE, lparam);
+    }
+}
+
+
+void LCPdatabase::loadDirectory(const boost::filesystem::path& p)
```

Comment:
There seem to be a problem when reading the LCP database on a Linux system with files such as `Read Me.txt` not being recognized as valid folder of file path

Reply:
@demoulinv Is it fixed?